### PR TITLE
Fix-Issues

### PR DESCRIPTION
**Pull request by ilikesymmetry**
Added item and station counters to help front-end with sectioning.  Changed latitude and longitude to floats instead of ints.  Added dining items field for eateries (applies only to non dining halls (ex: nasties, bust stop bagels, etc.)).  Incorporated image getting from the assets repo.

### CODE REVIEW

File: data.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 events = {}
 menus = {}
 items = {}
+dining_items = {}
```
Comment (young-k) [inline data.py]: this should be alphabetized

File: data.py [hunk 2]
```diff
@@ -31,7 +32,8 @@ def start_update():
         operating_hours=operating_hours,
         events=events,
         menus=menus,
-        items=items
+        items=items,
+        dining_items=dining_items
```
Comment (young-k) [inline data.py]: should be alphabetized as well

File: data.py [hunk 3]
```diff
@@ -42,30 +44,33 @@ def parse(data_json):
   global eateries
   for eatery in data_json['data']['eateries']:
     new_eatery = schema.EateryType(
+        about=eatery['about'],
+        about_short=eatery['aboutshort'],
+        campus_area=parse_campus_area(eatery['campusArea']),
+        coordinates=parse_coordinates(eatery['coordinates']),
+        dining_items=parse_dining_items(eatery['diningItems'], eatery['id']),
```
Comment (young-k) [inline data.py:50]: pt. 3

File: data.py [hunk 4]
```diff
@@ -42,30 +44,33 @@ def parse(data_json):
...
         name=eatery['name'],
         name_short=eatery['nameshort'],
-        about=eatery['about'],
-        about_short=eatery['aboutshort'],
-        image_url=parse_image_url(eatery['about']),
+        operating_hours=parse_operating_hours(eatery['operatingHours'], eatery['id']),
         payment_methods=parse_payment_methods(eatery['payMethods']),
+        phone=parse_phone(eatery['contactPhone'], eatery['name']),
+        slug=eatery['slug']
         # calender_id=eatery['googleCalenderId'],
```
Comment (young-k) [inline data.py]: delete this

File: data.py [hunk 5]
```diff
@@ -42,30 +44,33 @@ def parse(data_json):
...
-    image_url = found['src']
-    return image_url
-  print('Image url not found') # Should it be raise(FileNotFound()) exception?
-  return None
+def get_image_url(slug):
+  return constants.ASSET_BASE_URL + 'eatery-images/' + slug + '.jpg'
+
+def parse_phone(string, name):
+  if string is None:
+    print(name + ' missing phone number')
```
Comment (young-k) [inline data.py]: do we need print statements?

File: data.py [hunk 6]
```diff
@@ -42,30 +44,33 @@ def parse(data_json):
...
+    return 'N/A'
```
Comment (young-k) [inline data.py:69]: should be defined as a constant at the top

File: data.py [hunk 7]
```diff
@@ -94,12 +99,14 @@ def parse_events(event_list, eatery_id):
...
     new_event = schema.EventType(
-        description=event['descr'],
-        menu=parse_food_stations(event['menu'], eatery_id),
         start_time=event['start'],
         end_time=event['end'],
-        cal_summary=event['calSummary']
+        cal_summary=event['calSummary'],
+        description=event['descr'],
+        menu=stations,
+        station_count=len(stations)
```
Comment (young-k) [inline data.py:106]: alphabetize

File: data.py [hunk 8]
```diff
@@ -109,10 +116,12 @@ def parse_food_stations(station_list, eatery_id):
   global menus
   new_stations = []
   for station in station_list:
+    station_items = parse_food_items(station['items'], eatery_id)
     new_station = schema.FoodStationType(
         category=station['category'],
         sort_idx=station['sortIdx'],
-        items=parse_food_items(station['items'], eatery_id)
+        items=station_items,
+        item_count=len(station_items)
```
Comment (young-k) [inline data.py]: same thing

File: data.py [hunk 9]
```diff
@@ -131,6 +140,21 @@ def parse_food_items(item_list, eatery_id):
...
   return new_food_items
 
+def parse_dining_items(item_list, eatery_id):
+  global dining_items
+  new_dining_items = []
+  for item in item_list:
+    new_dining_item = schema.DiningItemType(
+        description=item['descr'],
+        category=item['category'],
+        item=item['item'],
```
Comment (young-k) [inline data.py]: alphabetize this as well

File: schema.py [hunk 10]
```diff
@@ -7,6 +7,7 @@ class Data(object):
   events = {}
   menus = {}
   items = {}
+  dining_items = {}
```
Comment (young-k) [inline schema.py]: same thing

File: schema.py [hunk 11]
```diff
@@ -15,10 +16,11 @@ def update_data(**kwargs):
     Data.events = kwargs.get('events')
     Data.menus = kwargs.get('menus')
     Data.items = kwargs.get('items')
+    Data.dining_items = kwargs.get('dining_items')
```
Comment (young-k) [inline schema.py]: same thing

File: schema.py [hunk 12]
```diff
@@ -32,6 +34,13 @@ class PaymentMethodsType(ObjectType):
   cornell_card = Boolean(required=True)
   mobile = Boolean(required=True)
 
+class DiningItemType(ObjectType):
+  description = String(required=True)
+  category = String(required=True)
+  item = String(required=True)
+  healthy = Boolean(required=True)
+  show_category = Boolean(required=True)
```
Comment (young-k) [inline schema.py:42]: +1

File: schema.py [hunk 13]
```diff
@@ -41,33 +50,38 @@ class FoodStationType(ObjectType):
...
+  item_count = Int(required=True)
 
 class EventType(ObjectType):
-  description = String(required=True)
-  menu = List(FoodStationType, required=True)
   start_time = String(required=True)
   end_time = String(required=True)
   cal_summary = String(required=True)
+  description = String(required=True)
+  menu = List(FoodStationType, required=True)
```
Comment (young-k) [inline schema.py:59]: again

File: schema.py [hunk 14]
```diff
@@ -41,33 +50,38 @@ class FoodStationType(ObjectType):
...
-  id = Int(required=True)
-  slug = String(required=True)
-  name = String(required=True)
-  name_short = String(required=True)
   about = String(required=True)
   about_short = String(required=True)
+  campus_area = Field(CampusAreaType, required=True)
+  coordinates = Field(CoordinatesType, required=True)
+  dining_items = List(DiningItemType, required=True)
+  eatery_type = String(required=True)
```
Comment (young-k) [inline schema.py:74]: keep it going

File: schema.py [hunk 15]
```diff
@@ -41,33 +50,38 @@ class FoodStationType(ObjectType):
...
   location = String(required=True)
+  name = String(required=True)
+  name_short = String(required=True)
   operating_hours = List(OperatingHoursType, required=True)
-  coordinates = Field(CoordinatesType, required=True)
-  campus_area = Field(CampusAreaType, required=True)
+  payment_methods = Field(PaymentMethodsType, required=True)
+  phone = String(required=True)
+  slug = String(required=True)
+  # calendar_id = String(required=True) # for scraping
```
Comment (young-k) [inline schema.py]: remove, and alphabetize

**Review by young-k — CHANGES_REQUESTED** [review]
Minor changes as always!

File: data.py (hunk 5 above)
Comment (ilikesymmetry) [reply data.py]: no but its nice to see who doesn't have numbers

**Comment by ilikesymmetry** [comment]
@young-k

**Review by young-k — APPROVED** [review]
Fix hound complaints, everything else lgtm

**Comment by kq-li** [comment]
hound has some weird complaints here -- the two newlines thing and the sort of short line maximum come to mind -- can we configure it? @young-k

**Pull request merged**